From 396e6c2c4f5d25328f4e0e38a7f69c096b80ace6 Mon Sep 17 00:00:00 2001
From: Josh Boyer <jwboyer@fedoraproject.org>
Date: Mon, 11 Nov 2013 08:39:16 -0500
Subject: [PATCH 20/34] OBSOLETE: FEDORA: lib/cpumask: Make CPUMASK_OFFSTACK
 usable without debug dependency

When CPUMASK_OFFSTACK was added in 2008, it was dependent upon
DEBUG_PER_CPU_MAPS being enabled, or an architecture could select it.
The debug dependency adds additional overhead that isn't required for
operation of the feature, and we need CPUMASK_OFFSTACK to increase the
NR_CPUS value beyond 512 on x86.  We drop the current dependency and make
sure SMP is set.

Bugzilla: N/A
Upstream-status: Nak'd, supposedly replacement coming to auto-select

Signed-off-by: Josh Boyer <jwboyer@fedoraproject.org>

Link: https://src.fedoraproject.org/rpms/kernel/c/9742e529b878106143a0a68c4734af4328688a99?branch=rawhide
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 lib/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Kconfig b/lib/Kconfig
index ee365b7402f1..16d1c4f95c5a 100644
--- a/lib/Kconfig
+++ b/lib/Kconfig
@@ -532,7 +532,8 @@ config CHECK_SIGNATURE
 	bool
 
 config CPUMASK_OFFSTACK
-	bool "Force CPU masks off stack" if DEBUG_PER_CPU_MAPS
+	bool "Force CPU masks off stack"
+	depends on SMP
 	help
 	  Use dynamic allocation for cpumask_var_t, instead of putting
 	  them on the stack.  This is a bit more expensive, but avoids
-- 
2.47.0

